src/components/pages/events/eventOverview/eventOverview.tsx

```diff
@@ -87,10 +113,48 @@ const EventOverview: React.FC = () => {
     }
   }, [eventContent, joinedErrorMsg, pastErrorMsg, addToast]);
 
-  useEffect(() => {
-    fetchEvents();
-  }, []);
+  const handleNextPage = () => {
+    if (currentPage < totalPages) {
+      setSkip((prevSkip) => prevSkip + limit);
+    }
+  };
+
+  const handlePreviousPage = () => {
+    if (currentPage > 1) {
+      setSkip((prevSkip) => Math.max(prevSkip - limit, 0));
+    }
+  };
 
+  const generatePages = () => {
+    const pages = [];
+
+    // Always show the first page
+    pages.push(1);
+
+    // Show ellipsis if there are several pages between the first and the current page
+    if (currentPage > 3) {
+      pages.push('...');
+    }
+
+    // Add pages around the current page
+    for (
+      let i = Math.max(2, currentPage - 1);
+      i <= Math.min(totalPages - 1, currentPage + 1);
+      i++
+    ) {
+      pages.push(i);
```

Comment:
In my opinion, this function would be more readable if the items pushed to the `pages` list was always of the same data type. So instead of `pages.push(i)` do `pages.push(${i})`. (this is nitpicking though)